src/server/services/graph/createConversionArrays.js

```diff
@@ -0,0 +1,104 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const Unit = require('../../models/Unit');
+const { getPath } = require('./createConversionGraph');
+const { pathConversion } = require('./pathConversion');
+
+/**
+ * For each unit in the list, assigns its index to the conversion table.
+ * For source units, the index is the row id.
+ * For destination units, the index is the column id.
+ * @param {*} units The list of units.
+ * @param {*} conn The connection to use.
+ */
+async function assignIndex(units, conn) {
+	let id = 0;
+	// The old indices may cause conflicts while we are assigning new ones.
+	// Therefore, we need to reset them to null first.
+	for (let unit of units) {
+		if (unit.unitIndex !== null) {
+			unit.unitIndex = null;
+			await unit.update(conn);
+		}
+	}
+	// Assigns new indices.
+	for (let unit of units) {
+		unit.unitIndex = id;
+		id += 1;
+		await unit.update(conn);
+	}
+}
+
+/**
+ * Returns the Cik array.
```

Comment:
Maybe: Returns the Cik which gives the slope, intercept and suffix name between each meter and unit where it is NaN, Nan, '' if no conversion.

The goal is for the JSDoc to tell what is going on without looking at code.